src/views/FarmAuction/components/AuctionHistory.tsx

```diff
@@ -75,20 +75,18 @@ const AuctionHistory: React.FC<AuctionHistoryProps> = ({ mostRecentClosedAuction
   const endDate = selectedAuction ? format(selectedAuction.auction.endDate, 'MMM. dd yyyy, hh:mm aa') : null
 
   const handleHistoryAuctionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const {
-      target: { value },
-    } = event
-    if (value) {
-      let newAuctionId = value
-      if (parseInt(value, 10) <= 0) {
-        newAuctionId = ''
-      }
-      if (parseInt(value, 10) >= mostRecentClosedAuctionId) {
-        newAuctionId = mostRecentClosedAuctionId.toString()
-      }
+    if (event.currentTarget.validity.valid) {
+      const {
+        target: { value },
+      } = event
+      const valueAsNumber = +value
+      const newAuctionId =
+        valueAsNumber >= mostRecentClosedAuctionId
```

Comment:
if we have `max` in input, no need for this line

also get the number type value by `event.target.valueAsNumber` should do the job

Reply:
I modified the code to use event.target.valueAsNumber. But since we need to use type=text for the reason that i explained above I think we should keep that line.

Reply:
I tested with valueAsNumber It always comes NaN. I reverted it back